### PR TITLE
Add snapcraft metadata

### PR DESCRIPTION
**Pull request by casept**
This is more-or-less the same stuff as in my unofficial zeal snap repo. The manifest has been adapted to use cmake. It needs testing on other machines.

### CODE REVIEW

**Comment by trollixx** [comment]
Thanks! I'll try to test this before merging and plugging to build.snapcraft.io.

Another question, is it possible to have daily builds pushed to edge, while only tagged releases go to beta/stable?

**Comment by casept** [comment]
> daily builds pushed to edge

AFAIK it's not possible to make `build.snapcraft.io` build on a schedule or when commits are made to a different repo than the one hosting the `snapcraft.yml` file. I think the "least hackish" option would be simply putting the snapcraft metadata into the main repo. That way a build and push to `edge` would be triggered by each new commit.

It would also make the `snapcraft.yml` easier to maintain. The `pull` step currently downloads a tagged release from GH, so building nightly builds would require a separate `snapcraft.yml` in a separate `packaging-snap-nightly` repo, whereas by letting the `snapcraft.yml` live in the main repo it could simply pull the source code from the directory it's in.

It would also ensure that if, for example, a user wishes to checkout an older version of zeal and build a snap package he would always have a working `snapcraft.yml` file for that particular point in history (because it's checked in alongside the source code). With the current model of storing the file separately the user would have to dig through this repo's history to hopefully find a version of the file that works with his particular commit in the main repo.

Another option might be to push an "empty" commit to this repo each time a nightly build should be pushed out, but that would require manual intervention or somewhat complex automation (and a server/CI service to run the automation on).

I'll ask on the snapcraft forum whether anyone knows a better workaround.

> tagged releases go to beta/stable

I'm not aware of any way to automate that using `build.snapcraft.io`, I think a 3rd party CI service with `snapcraft` support (or manually labelling a snap build as stable) would be needed to accomplish this.

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - libxslt1-dev
+      - libqt5x11extras5-dev
+      - libqt5webkit5-dev
+      - libsqlite3-dev
+      - extra-cmake-modules
+    stage-packages:
+      - xcb
+      - libarchive13
+      - libxcb-cursor0
+      - libqt5sql5-sqlite
```
Comment (trollixx) [inline snapcraft.yaml]: Zeal doesn't use `libqt5sql5-sqlite` .

File: snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  zeal:
+    plugin: cmake
+    source: https://github.com/zealdocs/zeal/archive/v$SNAPCRAFT_PROJECT_VERSION.tar.gz
+    build-packages:
+      - build-essential
+      - libarchive-dev
+      - libappindicator-dev
+      - libxcb-keysyms1-dev
+      - zlib1g-dev
```
Comment (trollixx) [inline snapcraft.yaml:23]: Do all dependencies need to be listed explicitly? I am pretty sure some of these depend on others already.

File: snap/.snapcraft/state [hunk 3]
```diff
@@ -0,0 +1,4 @@
+!GlobalState
```
Comment (trollixx) [inline snap/.snapcraft/state]: Isn't this file automatically generated?

**Review by trollixx — CHANGES_REQUESTED** [review]
Almost forgot about this, sorry. A few observations.

I wonder if IDE integration works with all isolation...

**Comment by trollixx** [comment]
Thanks! Sorry for the delay. Will try to hook this into snapcraft soon.

**Pull request merged**